src/adagrams.js

```diff
@@ -1,15 +1,133 @@
+const generatePoolArray = (LETTER_POOL) => {
+  let poolArray = [];
+  for(let letter in LETTER_POOL){
+    for(let i = 0; i<LETTER_POOL[letter]; i++){
+      poolArray.push(letter);
+    }
+  }
+  return poolArray;
+};
+
 export const drawLetters = () => {
   // Implement this method for wave 1
+    const LETTER_POOL = {
+    A: 9,
+    B: 2,
+    C: 2,
+    D: 4,
+    E: 12,
+    F: 2,
+    G: 3,
+    H: 2,
+    I: 9,
+    J: 1,
+    K: 1,
+    L: 4,
+    M: 2,
+    N: 6,
+    O: 8,
+    P: 2,
+    Q: 1,
+    R: 6,
+    S: 4,
+    T: 6,
+    U: 4,
+    V: 2,
+    W: 2,
+    X: 1,
+    Y: 2,
+    Z: 1,
+  };
+
+  let poolArray = generatePoolArray(LETTER_POOL);
+
+  const HANDSIZE = 10;
+  let hand = [];
+  let handDict = {};
+  for(let i = 0; hand.length < HANDSIZE; i++){
+    let randomNum = Math.floor(Math.random() * (poolArray.length));
+    let letter = poolArray[randomNum];
+
+    if(handDict[letter] === undefined || handDict[letter] < LETTER_POOL[letter]){
+      hand.push(letter);
+      if(handDict[letter] === undefined){
+        handDict[letter] = 1;
+      }else{
+        handDict[letter]++;
+      }
+    }
+  }
+  return hand;  
 };
 
 export const usesAvailableLetters = (input, lettersInHand) => {
   // Implement this method for wave 2
+  for(let i = 0; i < input.length; i++){
+    let letterIndex = lettersInHand.indexOf(input[i]);
+    if (letterIndex === -1){
+      return false;
+    }
+    lettersInHand.splice(letterIndex, 1);
+  }
+  return true;
 };
 
 export const scoreWord = (word) => {
   // Implement this method for wave 3
+  const calcScore =(word)=>{
```

Comment:
Did you intend to nest `calcScore` inside `scoreWord`?

We don't need `calcScore` to be an inner function so I'm inclined to say that it should be written outside of `scoreWord`.

Also, nitpick: spacing

```suggestion
  const calcScore = (word)=>{
```